### ISSUE TITLE
enum34 requirement causes issues

### ISSUE THREAD

**Issue by Phxntxm** [smartsheet-platform/smartsheet-python-sdk#122]
So I have a bit of a strange issue here, and it stems from the requirement of enum34. I am trying to use sshuttle for python 3.6, as well as this library...and doing so is incompatible due to the requirement mentioned above...the reason I bring this up here is because technically, I don't believe this library `smarthseet-python-sdk` should be considered compatible with python3.6. I'll explain this below...

The standard library `enum` introduced a new flag `IntFlag` for version 3.6, [the library re uses this in version 3.6](https://github.com/python/cpython/blob/master/Lib/re.py#L143). Due to this, if I force the primary path to import from to be the site packages installed (sshuttle does this, and it is required for its usage) then `enum34` takes over instead of the stdlib `enum`...and causes `re` to not be importable. Example of this is below:

```
dhess@dhess-laptop:~$ python3.6 -m pip show smartsheet-python-sdk
Name: smartsheet-python-sdk
Version: 1.3.3
Summary: Library that uses Python to connect to Smartsheet services (using API 2.0).
Home-page: http://smartsheet-platform.github.io/api-docs/
Author: Smartsheet
Author-email: api@smartsheet.com
License: Apache-2.0
Location: /usr/local/lib/python3.6/dist-packages
Requires: certifi, enum34, six, requests-toolbelt, requests, python-dateutil
Required-by:

dhess@dhess-laptop:~$ PYTHONPATH=/usr/local/lib/python3.6/dist-packages/ python3.6 -c "import re"
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.6/re.py", line 142, in <module>
    class RegexFlag(enum.IntFlag):
AttributeError: module 'enum' has no attribute 'IntFlag'
Error in sys.excepthook:
Traceback (most recent call last):
  File "/usr/lib/python3/dist-packages/apport_python_hook.py", line 53, in apport_excepthook
    if not enabled():
  File "/usr/lib/python3/dist-packages/apport_python_hook.py", line 24, in enabled
    import re
  File "/usr/lib/python3.6/re.py", line 142, in <module>
    class RegexFlag(enum.IntFlag):
AttributeError: module 'enum' has no attribute 'IntFlag'

Original exception was:
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/usr/lib/python3.6/re.py", line 142, in <module>
    class RegexFlag(enum.IntFlag):
AttributeError: module 'enum' has no attribute 'IntFlag'
```
This is because, as I mentioned, it's using enum34's file:
```
dhess@dhess-laptop:~$ PYTHONPATH=/usr/local/lib/python3.6/dist-packages/ python3.6 
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import enum      
>>>enum.__file__
'/usr/local/lib/python3.6/dist-packages/enum/__init__.py'
```
Instead of how it would act normally:
```
dhess@dhess-laptop:~$ python3.6
Python 3.6.7 (default, Oct 22 2018, 11:32:17) 
[GCC 8.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import enum
>>> enum.__file__
'/usr/lib/python3.6/enum.py'
```

In summary, this issue is caused by the combination of the `enum34` requirement by this package, as well as needing to set PYTHONPATH. 

From what I understand, enum34 is just for the purpose of being available for python2.7 correct? Can this be changed? Do you have any other suggestions? Currently what I'm doing is just uninstalling enum34, and will uninstall it anytime I update this library.

**Comment by urbandw** [comment]
This also [breaks autocompletion](https://github.com/tensorflow/tensorflow/issues/15136) other editor functionality in the Spyder IDE. It would be great to get rid of this requirement.

**Comment by timwellswa** [comment]
Merged into my development tree. Will pick up the change with the next update (should be pretty quick). Thanks Dan.
